File: python/atbash-cipher/atbash_cipher.py
from textwrap import wrap
from string import punctuation

CODE = {
    'a': 'z',
    'b': 'y',
    'c': 'x',
    'd': 'w',
    'e': 'v',
    'f': 'u',
    'g': 't',
    'h': 's',
    'i': 'r',
    'j': 'q',
    'k': 'p',
    'l': 'o',
    'm': 'n',
    'n': 'm',
    'o': 'l',
    'p': 'k',
    'q': 'j',
    'r': 'i',
    's': 'h',
    't': 'g',
    'u': 'f',
    'v': 'e',
    'w': 'd',
    'x': 'c',
    'y': 'b',
    'z': 'a',
}


def encode(plain_text):
    plain_text = plain_text.lower()
    stripped = ''.join(c for c in plain_text if c not in punctuation)
    stripped = stripped.replace(' ', '')

    result = ''
    for c in stripped:
        if c in CODE:
            result += CODE[c]
        else:
            result += c

    return ' '.join(wrap(result, 5))


def decode(ciphered_text):
    return encode(ciphered_text).replace(' ', '')

File: elixir/atbash-cipher/atbash_cipher.exs
defmodule Atbash do
  @doc """
  Encode a given plaintext to the corresponding ciphertext

  ## Examples

  iex> Atbash.encode("completely insecure")
  "xlnko vgvob rmhvx fiv"
  """
  @spec encode(String.t) :: String.t
  def encode(plaintext) do
    plaintext
    |> String.downcase
    |> String.replace(~r/[[:punct:]]|\s/, "")
    |> to_charlist
    |> encode([])
    |> Enum.chunk_every(5)
    |> Enum.join(" ")
  end

  @spec encode(charlist, charlist) :: charlist
  def encode([], acc), do: acc
  def encode([h|t], acc) do
    encoded =
      cond do
        h in ?a..?z -> 219 - h
        true -> h
      end
    encode(t, acc ++ [encoded])
  end

  @spec decode(String.t) :: String.t
  def decode(cipher) do
    encode(cipher)
    |> String.replace(" ", "")
  end
end
